### DISCUSSION TITLE
Error when compiling Méric-Cailletaud finite-strain Mfront file from the tutorial in MFront gallery

### DISCUSSION THREAD

**Discussion by selshawish** [thelfer/tfel#406] [Q&A] [answered]
Hi,

I started learning MFront by looking at MFront gallery and examples provided there. While trying to compile the MFront file for Méric-Cailletaud single crystal behavior at finite strain, I get the following error:

> mfront --obuild --interface=umat MericCailletaudFiniteStrainSingleCrystalViscoPlasticity.mfront

> Error while treating file 'MericCailletaudFiniteStrainSingleCrystalViscoPlasticity.mfront'
> BehaviourDSLBase::analyse: error while treating keyword '@Brick' at line '15' of file 'MericCailletaudFiniteStrainSingleCrystalViscoPlasticity.mfront'.
> BehaviourBrickBase::checkOptionsNames: brick 'FiniteStrainSingleCrystal' does not expect option 'poisson_ratio12'

The MFront file can be found here:
https://tfel.sourceforge.net/gallery/finitestrainsinglecrystal/MericCailletaudFiniteStrainSingleCrystalViscoPlasticity.mfront

The behavior is introduced here:
https://tfel.sourceforge.net/MericCailletaudSingleCrystalPlasticity.html

I'm using the MFront version 3.2.6.

Thank you for your consideration.

S.

**Comment by thelfer** [comment, marked as answer]
Dear @selshawish,

The syntax allowing to define the elastic properties in the definition of the brick has been introduced in more recent versions of TFEL.

You can use something like this:

```cxx
@Brick FiniteStrainSingleCrystal;
@RequireStiffnessOperator;
```

However, I truly recommand that you update your MFront versions !
